### PR TITLE
Enhancement | Remove empy provider configuration block warning

### PR DESCRIPTION
**Pull request by angelofenoglio**
## What
* Update module to fix "Empty provider configuration block" warning.

## Why
To get rid of the message:
```
╷
│ Warning: Empty provider configuration blocks are not required
│ 
│   on .terraform/modules/terraform_backend/config.tf line 5:
│    5: provider "aws" {
│ 
│ Remove the aws.main_region provider block from module.terraform_backend. Add aws.main_region to the list of configuration_aliases for aws in required_providers to define the provider
│ configuration name.
│ 
│ (and one more similar warning elsewhere)
╵
```
when using this module.

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -1,8 +1,12 @@
 
 terraform {
-  required_version = ">= 0.13.2"
+  required_version = ">= 1.0.9"
 
   required_providers {
-    aws = "~> 3.0"
+    aws = {
```
Comment (diego-ojeda-binbash) [inline versions.tf:6]: @angelofenoglio  So the fix is only this piece here, right?

**Review by exequielrafaela — APPROVED** [review]

**Review by exequielrafaela — COMMENTED** [review]
@angelofenoglio the PR looks very good :) 

Small comment regarding CI failing 

I think you've already fixed this in another repo
- https://app.circleci.com/pipelines/github/binbashar/terraform-aws-tfstate-backend/118/workflows/0aa4584f-b7fc-4d46-8796-c443739ece82/jobs/214

```
pre-commit run --all-files
[INFO] Initializing environment for git://github.com/pre-commit/pre-commit-hooks.
An unexpected error has occurred: CalledProcessError: command: ('/usr/bin/git', 'fetch', 'origin', '--tags')
return code: 128
expected return code: 0
stdout: (none)
stderr:
    fatal: remote error: 
      The unauthenticated git protocol on port 9418 is no longer supported.
    Please see https://github.blog/2021-09-01-improving-git-protocol-security-github/ for more information.
    
Check the log at /home/circleci/.cache/pre-commit/pre-commit.log
make: *** [@bin/makefiles/terratest14/terratest14.mk:84: pre-commit] Error 3

Exited with code exit status 2
CircleCI received exit code 2
```

Regarding tests here https://github.com/binbashar/terraform-aws-tfstate-backend/blob/master/tests/verify_output_test.go we'll need to update the examples to cover the new module version
- https://app.circleci.com/pipelines/github/binbashar/terraform-aws-tfstate-backend/118/workflows/0aa4584f-b7fc-4d46-8796-c443739ece82/jobs/215

So if this will stop you from delivering your work please leave an issue in the repo and we can pull it as soon as possible. In general is very important to consider the tests effort, but we can do an exception this time.

CC: @binbashar/leverage-project-terraform-admin @binbashar/leverage-project-terraform-dev

File: versions.tf (hunk 1 above)
Comment (angelofenoglio) [reply versions.tf:6]: Essentially yes, but all the references need to be updated also.

Comment (angelofenoglio) [reply versions.tf:6]: Seems reasonable, I'll make these changes then!

Comment (diego-ojeda-binbash) [reply versions.tf:6]: Oh, is that right? I would've thought that the following is implicit:
```
resource "foo" "bar" {
    ...
    provider = aws
    ...
}
```
And thus it does not need to be explicitly defined.

**Pull request merged**